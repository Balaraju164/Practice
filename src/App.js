import React, { Component, Fragment } from 'react';
// import YourLocation from './Zemoso_components/YourLocation'
// import Slick from './Components/Slick/slick'
//import Jest from './Components/Jest_examples';
import Form from './Components/FormProject/Form';

export class App extends Component {
  render() {
    return (
      <Fragment>
        <Form />
      </Fragment>
    );
  }
}

export default App;
