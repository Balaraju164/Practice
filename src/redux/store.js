import { applyMiddleware, combineReducers } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
//import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import { countChange } from './reducers/countChange';

const reducer = combineReducers({
  count: countChange,
});

const initialstate = {};

//const middleware=[thunk]

const store = configureStore(
  { reducer },
  initialstate,
  composeWithDevTools(applyMiddleware())
);

export default store;
