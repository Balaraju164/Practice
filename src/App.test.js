import { render, cleanup, fireEvent, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import { combineReducers, applyMiddleware } from 'redux';
// import { composeWithDevTools } from 'redux-devtools-extension';
import { initialState, countChange } from './redux/reducers/countChange';
import App from './App';
import YourLoaction from './Zemoso_components/YourLocation';
import Jest, { BASE_URL, fetchUsers } from './Components/Jest_examples';
import axios from 'axios';

afterEach(cleanup);

test('getsnapshot', () => {
  const { asFragment } = render(<YourLoaction />);
  //const linkElement = getByText("Hello");
  expect(asFragment(<YourLoaction />)).toMatchSnapshot();
});

test('test content', () => {
  const { getByText } = render(<YourLoaction />);
  const linkElement = getByText(
    'More than 2000 people are using Green Commute'
  );
  expect(linkElement.textContent).toBe(
    'More than 2000 people are using Green Commute'
  );
});

const reducer = combineReducers({
  count: countChange,
});

const renderWithRedux = (
  component,
  { initialState, store = configureStore({ reducer }, initialState) } = {}
) => {
  return {
    ...render(<Provider store={store}>{component}</Provider>),
    store,
  };
};

test('test redux', () => {
  const { getByTestId, getByText } = renderWithRedux(<Jest />);
  fireEvent.click(getByText('Down'));
  expect(getByTestId('counter')).toHaveTextContent('-1');
});

test('test countchange reducer', () => {
  const count1 = { count: 1 };
  expect(countChange({ count: 0 }, { type: 'INCREMENT' })).toEqual(count1);
});

jest.mock('axios');

describe('fetchUsers', () => {
  describe('when API call is successful', () => {
    it('should return users list', async () => {
      const users = {
        data: [
          { id: 1, name: 'John' },
          { id: 2, name: 'Andrew' },
        ],
      };
      axios.get.mockResolvedValueOnce(users);
      const result = await fetchUsers();
      expect(axios.get).toHaveBeenCalledWith(`${BASE_URL}/users`);
      expect(result).toEqual(users);
    });
  });

  describe('when API call fails', () => {
    it('should return empty users list', async () => {
      const message = 'Network Error';
      axios.get.mockRejectedValueOnce(new Error(message));

      // when
      const result = await fetchUsers();

      // then
      expect(axios.get).toHaveBeenCalledWith(`${BASE_URL}/users`);
      expect(result).toEqual([]);
    });
  });
});

// test('firing events',()=>{
//   const {getByText,getByTestId} =render(<Jest/>)
//   fireEvent.click(getByText('Down'))
//   expect(getByTestId('counter')).toHaveTextContent('-1')
// })

// test('test settimeout',async()=>{
//   const {getByText,getByTestId} =render(<Jest/>)
//   //fireEvent.click(getByText('Increment'))
//   await waitFor(() => {
//     expect(getByText('1')).toBeInTheDocument();
//   })
//   //expect(counter).toHaveTextContent('1')
// })
