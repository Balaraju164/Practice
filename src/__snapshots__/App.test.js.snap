// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getsnapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="row1"
      >
        <div
          class="content"
        >
          <nav
            class="navbar"
          >
            <a
              href="#"
            >
              <p
                class="number"
              >
                1
              </p>
              Your location
            </a>
            <a
              href="#"
            >
              <p
                class="number"
              >
                2
              </p>
              Job Location
            </a>
            <a
              class="disabled"
              href="#"
            >
              <p
                class="number1"
              >
                3
              </p>
              Your skills
            </a>
          </nav>
          <div
            class="data_content"
          >
            <h3>
              More than 2000 people are using Green Commute
            </h3>
            <button
              class="back_button"
              type="button"
            >
              <i
                class="fas fa-arrow-left"
              />
              back
            </button>
            <p>
              Where do you want to work?
            </p>
            <input
              placeholder="enter you job location"
              type="text"
            />
            <div
              class="buttons"
            >
              <button
                class="butto"
                type="button"
              >
                Next
              </button>
              <button
                class="butto"
                type="button"
              >
                Skip
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row2"
      >
        <img
          alt="location_image"
          src="/images/zemoso_image2.jpg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
